tidy3d/plugins/adjoint/components/data/data_array.py

```diff
@@ -312,28 +313,37 @@ def isel(self, **isel_kwargs) -> JaxDataArray:
 
         return self_sel
 
-    def sel(self, indexers: dict = None, method: str = "nearest", **sel_kwargs) -> JaxDataArray:
+    def sel(self, indexers: dict = None, method: str | None = None, **sel_kwargs) -> JaxDataArray:
         """Select a value from the :class:`.JaxDataArray` by indexing into coordinate values."""
+        if method is None:
+            return self._sel_exact(sel_kwargs)
+        elif method == "nearest":
+            return self._sel_nearest(sel_kwargs)
+        else:
+            raise NotImplementedError(f"Unkown selection method: {method}.")
+
+    def _sel_exact(self, sel_kwargs: dict) -> JaxDataArray:
         isel_kwargs = {}
-        for coord_name, sel_kwarg in sel_kwargs.items():
-            coord_list = self.get_coord_list(coord_name)
-            if isinstance(sel_kwarg, (tuple, list, np.ndarray)):
-                sel_kwarg = list(sel_kwarg)
-                isel_kwargs[coord_name] = []
-                for _sel_kwarg in sel_kwarg:
-                    if _sel_kwarg not in coord_list:
-                        raise DataError(
-                            f"Could not select '{coord_name}={_sel_kwarg}', value not found."
-                        )
-                    coord_index = coord_list.index(_sel_kwarg)
-                    isel_kwargs[coord_name].append(coord_index)
-            else:
-                if sel_kwarg not in coord_list:
-                    raise DataError(
-                        f"Could not select '{coord_name}={sel_kwarg}', value not found."
-                    )
-                coord_index = coord_list.index(sel_kwarg)
-                isel_kwargs[coord_name] = coord_index
+        for coord_name, vals in sel_kwargs.items():
+            coord_list = np.asarray(self.get_coord_list(coord_name))
+            vals = np.atleast_1d(vals)
+            dist = np.abs(coord_list[None] - vals[:, None])
```

Comment:
Similar comment below but trying to understand the idea here, so `coord_list[None]` gives an array shaped `(1, M)` and then `vals[:, None]` gives `(1, N)`, we take the difference and get an array shaped `(N, M)`.

The next line, by applying `np.where(...)[1]`, we select out all of the `N` indices into `coord_list`?

What's a little confusing is that in the nearest method, it seems the `dist` array is transposed compared to this one, it might be better to either keep them consistent, or probably just to refactor so that we compute this `dist` method for both `nearest` and `exact` and then have a separate logic for computing `indices` given the `dist` (for nearest and exact)?

Reply:
Yes, the reason for doing this was that it preserves the ordering of the coordinates this way. I'll have another look if I can do that without swapping the indices.